docs/rust_migration.md

```diff
@@ -0,0 +1,27 @@
+# Migrating from Older Versions
+
+The rust library is versioned according to the principles of [Semantic Versioning](https://semver.org). We define that the left-most non-zero component of the version is the major version, followed by the minor and optionally patch version. That means releases in the "0.y.z" series treat changes in "y" as a major release, which can contain incompatible API changes, while changes in just "z" are minor. For example the release 0.1.6 is fully backwards compatible to 0.1.5, but it contains new functionality. The release 0.2.0 however is a new major version compared to 0.1.x and may contain API incompatible changes.
+
+This guide lists all API incompatible changes between major versions and describes how you can migrate your application's source code.
+
+## Migrating from Version 0.1.x to 0.2.0
+
+In the 0.2.x series we have increased the minimum version of rust. You need to have rust compiler version >= 1.56 installed.
```

Comment:
this seems to assume that we will not raise it again in the 0.2.x series.
I'd leave that out, or use similar phrasing than in C++

Reply:
The C++ guide says this:

> In the 0.2.x series we have increased the minimum version of C++. You need to have a C++ compiler installed that supports C++ 20 or newer.

Does that need to get updated as well?

I do get your point though and will change this to read 0.2.0.

Reply:
I'll adapt it as I add the API change for C++.

Reply:
I updated that as well as the C++ docs which I copied the text from.

I needed to touch the C++ migration guide anyway to add the C++ side of things there.